### DISCUSSION TITLE
License conflict with Viper

### DISCUSSION THREAD

**Discussion by bhansalivikas** [spf13/viper#1436] [Q&A]
Hi Viper Team,

I see that Viper is MIT licensed and more flexible. However, it uses some components from github.com/hashicorp/ which are all MPL 2.0 licenses. This means if we use Viper to build any tool, we indirectly get a MPL copy-left license dependency which enforce us to make our code open-source. Requesting help on this licensing enforcements.

Regards

**Comment by sagikazarmark** [comment]
HI @vibhansa-msft 

> which enforce us to make our code open-source

I don't believe that assessment is correct. MPL specifically does NOT enforce you to open source your own code:

> However, a larger work using the licensed work may be distributed under different terms and without source code for files added in the larger work.

I'm not a lawyer though, so I can't say what your legal department would require you to do.

I'd suggest checking again if MPL is acceptable for you. If not, we'll figure something out.
